core/application/app_configuration.hpp

```diff
@@ -86,6 +86,11 @@ namespace kagome::application {
      */
     virtual const std::optional<std::string> &nodeKeyFile() const = 0;
 
+    /**
+     * @return true if generated libp2p networking key should be saved
+     */
+    virtual bool saveNodeKey() const = 0;
```

Comment:
I think the name is slightly misleading, it reads as if the method saved the node key. I suggest shouldSaveNodeKey or something alike.